From bb8740e4f0138e793c0b2e0fe42133e834e58456 Mon Sep 17 00:00:00 2001
From: Rahul Arya <aryarahul@google.com>
Date: Wed, 28 Sep 2022 20:54:29 +0000
Subject: [PATCH] Pass IRK rotation flag into native

Ignore-AOSP-First: Security fix
Test: Flashed build, set the prop and watched the stack restart
Bug: 195410559
Change-Id: I6496b34c144e697308109d5ee9528f4ae5c9b4b4
Merged-In: I6496b34c144e697308109d5ee9528f4ae5c9b4b4
(cherry picked from commit f7443b78ac56071c8dce2be865f96a566b51433b)
Merged-In: I6496b34c144e697308109d5ee9528f4ae5c9b4b4
---
 .../src/com/android/bluetooth/btservice/AdapterService.java   | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/android/app/src/com/android/bluetooth/btservice/AdapterService.java b/android/app/src/com/android/bluetooth/btservice/AdapterService.java
index f44b6d466c..7e50b5c8d4 100644
--- a/android/app/src/com/android/bluetooth/btservice/AdapterService.java
+++ b/android/app/src/com/android/bluetooth/btservice/AdapterService.java
@@ -5035,6 +5035,7 @@ public class AdapterService extends Service {
     private static final String GD_RUST_FLAG = "INIT_gd_rust";
     private static final String GD_LINK_POLICY_FLAG = "INIT_gd_link_policy";
     private static final String GATT_ROBUST_CACHING_FLAG = "INIT_gatt_robust_caching";
+    private static final String IRK_ROTATION_FLAG = "INIT_irk_rotation";
 
     /**
      * Logging flags logic (only applies to DEBUG and VERBOSE levels):
@@ -5091,6 +5092,9 @@ public class AdapterService extends Service {
                 GATT_ROBUST_CACHING_FLAG, false)) {
             initFlags.add(String.format("%s=%s", GATT_ROBUST_CACHING_FLAG, "true"));
         }
+        if (DeviceConfig.getBoolean(DeviceConfig.NAMESPACE_BLUETOOTH, IRK_ROTATION_FLAG, false)) {
+            initFlags.add(String.format("%s=%s", IRK_ROTATION_FLAG, "true"));
+        }
         if (DeviceConfig.getBoolean(DeviceConfig.NAMESPACE_BLUETOOTH,
                 LOGGING_DEBUG_ENABLED_FOR_ALL_FLAG, false)) {
             initFlags.add(String.format("%s=%s", LOGGING_DEBUG_ENABLED_FOR_ALL_FLAG, "true"));
-- 
2.38.1.273.g43a17bfeac-goog

